### PR TITLE
APR-57: add hosted OPD CLI config

### PR DESCRIPTION
**Pull request by tim0120**
## Summary
- allow Hosted Training configs to use `loss = "opd"` with `[teacher]`
- keep `loss = "rl"` as the generated template default while documenting OPD as an accepted option
- send OPD loss and teacher config through the existing `/rft/runs` payload

## Scope
This is CLI-only. It does not implement hosted/platform acceptance or teacher-logprob runtime wiring.

Ready for review, but merge should remain ordered behind the hosted API/runtime path for OPD.

## Verification
- `.venv/bin/python -m pytest packages/prime/tests/test_rl_config.py packages/prime/tests/test_rl_api.py` (`51 passed`)
- `.venv/bin/ruff format packages/prime/src/prime_cli/commands/rl.py packages/prime/tests/test_rl_api.py packages/prime/tests/test_rl_config.py`
- `.venv/bin/ruff check packages/prime/src/prime_cli/commands/rl.py packages/prime/tests/test_rl_api.py packages/prime/tests/test_rl_config.py`
- `git diff --check`

## Review Notes
- Codex CLI review found no actionable regressions
- Claude review could not run because the org had hit its monthly usage limit at the time of the earlier review

Linear: APR-57

---

> [!NOTE]
> **Low Risk**
> Small validation and documentation changes plus tests; no auth or runtime logic. Users could submit OPD runs before the backend supports them if merged ahead of the API.
> 
> **Overview**
> **Hosted Training CLI** now treats **`loss = "opd"`** like SFT for config and API: OPD is documented in the generated template, validation no longer rejects OPD as unsupported, and **`[teacher]` is required** for both `sft` and `opd` (still forbidden for `rl`).
> 
> Template comments were broadened from SFT-only distillation to **SFT/OPD** with shared teacher setup. Tests cover loading OPD TOML, rejecting OPD without a teacher, and asserting **`/rft/runs`** payloads include `loss: "opd"` and teacher config.
> 
> This is **CLI-only**; platform/runtime OPD support is assumed to land separately.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 961bcdb46669bd890d48dce283111715f522dd80. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by tim0120** [comment]
@codex review

**Comment by tim0120** [comment]
@codex review

**Comment by tim0120** [comment]
@codex review

File: packages/prime/src/prime_cli/commands/rl.py [hunk 1]
```diff
@@ -664,13 +664,8 @@ def validate_config_consistency(self) -> "RLConfig":
...
             raise ValueError("teacher can only be set when loss is 'sft' or 'opd'")
-        if self.loss == "sft" and self.teacher is None:
-            raise ValueError("teacher is required when loss is 'sft'")
-        if self.loss == "opd":
-            raise ValueError(
-                "loss='opd' is not supported for hosted runs yet; OPD requires "
-                "teacher logprob scoring support in the hosted runtime"
-            )
+        if self.loss in {"sft", "opd"} and self.teacher is None:
+            raise ValueError(f"teacher is required when loss is '{self.loss}'")
```
[the remark this answers is not in the document]
Comment (AmeenP) [reply packages/prime/src/prime_cli/commands/rl.py:696]: Keeping this CLI-only PR ready for review, but not independently mergeable ahead of the hosted runtime path. The PR body now calls out the merge ordering behind the platform OPD API/runtime stack (#2342/#3007/#3008). Platform remains responsible for accepting/running OPD and teacher logprob scoring; this PR only exposes the CLI config surface.
